server/src/socketHandlers/socketHandlers.ts

```diff
@@ -177,19 +177,45 @@ export default function initializeSocketHandler(io: Server) {
     });
 
     socket.on("start_game", ({ lobbyId, members }) => {
-      // Instantiate relevant classes
-      //const gameCards = initializeGameCards(members);
-      //const playerObjects = initializePlayers(members, gameCards);
-      //const dice = new Dice(SixSidedDie);
-      //game = new QwixxLogic(playerObjects, dice);
       lobbiesMap[lobbyId].startGame();
 
-      // Create path data and players' gameboard states to send back to client
-      const initialPlayersState = lobbiesMap[lobbyId].gameLogic?.players;
+      // Create path data and serialized gamelogic object to the front end
+      const gameLogic = lobbiesMap[lobbyId].gameLogic;
+      const serializedGameLogic = gameLogic ? gameLogic.serialize() : null;
       const path = `/game/${lobbyId}`;
-      const responseData = { path: path, players: initialPlayersState };
+      const responseData = { path: path, gameState: serializedGameLogic };
 
       io.to(lobbyId).emit("game_initialised", responseData);
+      console.log(serializedGameLogic);
+    });
+
+    socket.on("mark_numbers", ({ lobbyId, userId, playerChoice }) => {
+      const gameLogic = lobbiesMap[lobbyId].gameLogic;
+
+      if (!gameLogic) {
+        console.log(`Game logic not found for lobby ${lobbyId}`);
+        return;
+      }
+
+      const player = gameLogic.players.find((player) => player.name === userId);
+
+      if (!player) {
+        console.log(`Player ${userId} not found in lobby`);
+        return;
+      }
+
+      console.log(playerChoice);
+
+      const { row: rowColour, num } = playerChoice;
+      player.gameCard.markNumbers(rowColour, num);
+      console.log(player.gameCard.MarkedNumbers);
+
+      const serializedGameLogic = gameLogic ? gameLogic.serialize() : null;
+      const path = `/game/${lobbyId}`;
+      const responseData = { path: path, gameState: serializedGameLogic };
```

Comment:
Do we need to send a path again? We're sending the path when starting the game so that all players can be redirected to the game page, but I don't think we're sending players to a different page on every move.

Reply:
No we dont need this piece of code. It is not being used to redirect. The gamepath is being set in the state on the front end and will not really change unless they get disconnected or join a new lobby